config/configoauth2/configoauth2client.go

```diff
@@ -0,0 +1,69 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configoauth2
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"golang.org/x/oauth2"
```

Comment:
We use go-oidc in the `configauth` package. Do you know if we can use the same package in both places, or are they complimentary to each other?

Reply:
That is correct, these packages are complimentary, go-odic is built on x/oauth2 handling ODIC (which itself is a specification on top of OAuth2) specific tasks